pkg/azure/aks/machine_store.go

```diff
@@ -125,16 +141,44 @@ func (m *MachineStore) getVmInfoFromVmss(ctx context.Context, rgName, vmssName,
 				continue
 			}
 
-			vmInfo[vmName] = &VirtualMachineInfo{
-				Name:            vmName,
-				Region:          to.String(v.Location),
-				OwningVMSS:      vmssName,
-				OwningCluster:   cluster,
-				MachineTypeSku:  to.String(v.SKU.Name),
-				Priority:        priority,
-				OperatingSystem: osInfo,
+			if v.SKU == nil || v.SKU.Name == nil {
+				m.logger.LogAttrs(ctx, slog.LevelDebug, "no VM Sku found", slog.String("machineName", vmName))
+				continue
+			}
+			vmSku := to.String(v.SKU.Name)
+
+			vmRegion := to.String(v.Location)
+			if len(vmRegion) == 0 {
+				m.logger.LogAttrs(ctx, slog.LevelDebug, "no VM region found", slog.String("machineName", vmName))
+				continue
+			}
+
+			if v.Properties == nil || v.Properties.VMID == nil {
+				m.logger.LogAttrs(ctx, slog.LevelDebug, "no VM ID found", slog.String("machineName", vmName))
+				continue
 			}
-			m.logger.LogAttrs(ctx, slog.LevelDebug, "found machine information", slog.String("machineName", vmName))
+			vmId := to.String(v.Properties.VMID)
+
+			if vmSizeInfo, ok := m.MachineSizeMap[vmRegion][vmSku]; ok {
+				vmInfo[vmId] = &VirtualMachineInfo{
+					Name:            vmName,
+					Id:              vmId,
+					Region:          vmRegion,
+					OwningVMSS:      vmssName,
+					OwningCluster:   cluster,
+					MachineTypeSku:  vmSku,
+					Priority:        priority,
+					OperatingSystem: osInfo,
+
+					NumOfCores:     to.Int32(vmSizeInfo.NumberOfCores),
+					MemoryInMB:     to.Int32(vmSizeInfo.MemoryInMB),
```

Comment:
Do you know if this is MB or MiB? I don't think it impacts this specific PR, but it's important to make sure it aligns with whatever the price is.

Reply:
After looking up the docs, it looks to be in MiB because the docs list computation in GiB?

https://learn.microsoft.com/en-us/azure/virtual-machines/sizes/memory-optimized/e-family

<img width="776" alt="image" src="https://github.com/grafana/cloudcost-exporter/assets/33067676/e4d6b88c-422e-4dd2-a3b7-1525bd15d79a">

Good call out!

[The golang SDK docs, though... ](https://github.com/Azure/azure-sdk-for-go/blob/v12.5.0-beta/arm/compute/models.go#L2267) mention MB 😱 

Updated.